HippUnfoldHCP/PostHippUnfoldHCP.sh

```diff
@@ -0,0 +1,300 @@
+#!/bin/bash
+set -eu
+pipedirguessed=0
+if [[ "${HCPPIPEDIR:-}" == "" ]]
+then
+  # pipedirguessed=1
+   #fix this if the script is more than one level below HCPPIPEDIR
+   export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
+
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"
+
+
+opts_SetScriptDescription "Make some BIDS structures and run HippUnfold"
+
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "folder containing all subjects"
+opts_AddMandatory '--subject' 'Subject' 'subject ID' ""
+opts_AddOptional '--hippunfold-dir' 'PhysicalHippUnfoldDIR' 'path' "location of HippUnfold outputs"
+opts_AddOptional '--atlas-hippunfold-dir' 'AtlasHippUnfoldDIR' 'path' "location of Atlas HippUnfold outputs"
+
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+opts_ShowValues
+
+log_Msg "Starting PostHippUnfold pipeline for subject: $Subject"
+
+T1wFolder="$StudyFolder/$Subject/T1w"
+AtlasFolder="$StudyFolder/$Subject/MNINonLinear"
+
+if [ -z ${PhysicalHippUnfoldDIR} ] ; then
+  PhysicalHippUnfoldDIR="${T1wFolder}/HippUnfold"
+fi 
+
+if [ -z ${AtlasHippUnfoldDIR} ] ; then
+  AtlasHippUnfoldDIR="${AtlasFolder}/HippUnfold"
+fi 
+
+function PostHippUnfold {
+HippUnfoldFolderOut=${1}
+PhysicalHippUnfoldFolder=${2}
+AtlasHippUnfoldFolder=${3}
+T1wFolder=${4}
+AtlasFolder=${5}
+Subject=${6}
+Modality=${7}
+Mesh=${8}
+
+if [ ${Modality} = "T1wT2w" ] ; then
+  Modality="T1w"
+  Flag="On"
+else
+  Flag="Off"
+fi
+
+function PALETTE {
+File=${1}
```

Comment:
Probably better to define this function outside the body of the other function.  Nesting like this doesn't remove the function once the outer function ends (so it doesn't have the usual benefits of nested functions), but it does redundantly redefine it each call.  It also increases the distance between the start and end of the outer function.

Also, the body isn't indented relative to the context.  I don't want matlab-style indentation to leak outside of matlab (I don't even like it in matlab...).

Reply:
I don't think I was clear on what was inside or outside a function's scope, so I put things inside just to be sure.  Let's see what happens with the HippUnfold inputs...